inc/migration.class.php

```diff
@@ -28,96 +28,227 @@
  * -------------------------------------------------------------------------
  */
 
-class PluginFieldsMigration extends Migration {
-
-   function __construct($ver = "") {
-      parent::__construct($ver);
-   }
-
-   static function install(Migration $migration, $version) {
-      global $DB;
-
-      $fields_migration = new self;
-
-      if ($DB->tableExists("glpi_plugin_customfields_fields")) {
-         if (!$fields_migration->updateFromCustomfields()) {
-            return false;
-         }
-      }
-
-      return true;
-   }
-
-   static function uninstall() {
-      return true;
-   }
-
-   function updateFromCustomfields($glpi_version = "0.80") {
-      //TODO : REWRITE customfield update
-      return true;
-   }
-
-   function displayMessage($msg) {
-      Session::addMessageAfterRedirect($msg);
-   }
-
-   function migrateCustomfieldTypes($old_type) {
-      $types = [
-         'sectionhead' => 'header',
-         'general'     => 'text',
-         'money'       => 'text',
-         'note'        => 'textarea',
-         'text'        => 'textarea',
-         'number'      => 'number',
-         'dropdown'    => 'dropdown',
-         'yesno'       => 'yesno',
-         'date'        => 'date'
-      ];
-
-      return $types[$old_type];
-   }
-
-   /**
-    * Return SQL fields corresponding to given additionnal field.
-    *
-    * @param string $field_name
-    * @param string $field_type
-    *
-    * @return array
-    */
-   public static function getSQLFields(string $field_name, string $field_type): array {
-
-      $default_key_sign = DBConnection::getDefaultPrimaryKeySignOption();
-
-      $fields = [];
-      switch (true) {
-         case $field_type === 'header':
-            // header type is for read-only display purpose only and has no SQL field
-            break;
-         case $field_type === 'dropdown':
-         case preg_match('/^dropdown-.+/i', $field_type):
```

Comment:
`case preg_match('/^dropdown-.+/i', $field_type):` was a legitimate case and should be preserved.

Diff is huge here, and it is hard to validate the code. Unless I am wrong, if #528 is merged, only the new command and corresponding `PluginFieldsMigration` methods are required.

I would prefer serarate PR for PSR-12 migration and for migration of PHP template files to abstract classes (should be done globally). We could work on these later.

Reply:
Ok, let's merge #528 and then I'll open separate PR for each topics.

PS: for the huge diff, it's mostly whitespaces and you can ignore them:
![image](https://user-images.githubusercontent.com/42734840/171828689-b749db68-0e5c-4305-9426-77e4b8474caf.png)